### PR TITLE
log-writer: make max log length configurable

### LINKED ISSUE #208: Make the log output limit of 4mb configurable
Hi Hi

Several Enterprise customers have asked to be able to customize the log output limit from 4mb to 40mb.

Can we please make this limit configurable so customers can choose their own limit.

Thanks a bundle

Josh

### PR DESCRIPTION
**Pull request by henrikhodne**
Fixes #208.

### CODE REVIEW

**Comment by henrikhodne** [comment]
@meatballhat r?

File: processor.go [hunk 1]
```diff
@@ -46,12 +47,21 @@ type Processor struct {
...
 
+type ProcessorConfig struct {
+	Hostname            string
+	HardTimeout         time.Duration
+	LogTimeout          time.Duration
+	ScriptUploadTimeout time.Duration
+	StartupTimeout      time.Duration
+	MaxLogLength        int
+}
+
```
Comment (meatballhat) [inline processor.go]: :clap:

File: processor.go [hunk 2]
```diff
@@ -46,12 +47,21 @@ type Processor struct {
 	SkipShutdownOnLogTimeout bool
 }
 
+type ProcessorConfig struct {
+	Hostname            string
```
Comment (henrikhodne) [inline processor.go]: Oops, this field is never used.

Comment (emdantrim) [inline processor.go]: fixed!

**Pull request merged**